File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/houses/HousePropertiesMessage.scala





















// Generated on 05/03/2013 10:34:44
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait HousePropertiesMessage extends MessageDeserializer {
  val messageId = 5734

  protected trait Message extends BaseMessage with Serializable {
    val messageId = HousePropertiesMessage.messageId

    val properties: HouseInformations.TypeImpl

    def serialize(buf: Buffer) {
      buf.writeUShort(properties.typeId)
      properties.serialize(buf)
    }

  }

  protected def create(properties: HouseInformations.TypeImpl): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val properties = types.get[HouseInformations.TypeImpl](buf.readShort());

    create(properties)
  }

}

object HousePropertiesMessage extends HousePropertiesMessage {
  case class MessageImpl(properties: HouseInformations.TypeImpl) extends Message
  type Target = MessageImpl

  override def create(properties: HouseInformations.TypeImpl) = MessageImpl(properties)

  def apply(properties: HouseInformations.TypeImpl) = create(properties)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/HousePropertiesMessage.java


// Generated on 05/08/2013 19:37:46
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class HousePropertiesMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5734;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public HouseInformations properties;
    
    public HousePropertiesMessage() { }
    
    public HousePropertiesMessage(HouseInformations properties) {
        this.properties = properties;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(properties.getTypeId());
        properties.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        properties = ProtocolTypeManager.getInstance().build(buf.readShort());
        properties.deserialize(buf);
    }
    
}
